nx-X11/programs/Xserver/hw/nxagent/Reconnect.c

```diff
@@ -438,22 +438,10 @@ Bool nxagentReconnectSession(void)
 
   if (nxagentKeyboard != NULL)
   {
-    int size;
+    nxagentOldKeyboard = strndup(nxagentKeyboard, strlen(nxagentKeyboard));
+    /* FIXME: check if nxagentOldKeyboard == NULL and abort */
```

Comment:
Just use
```C
  FatalError ("strndup failed for the keyboard settings");
```
or the like. A failing `strndup ()` is pretty much a death sentence anyway, unless the string to be copied is unreasonable big. Which, frankly, would be a problem on its own...